android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/PackageActivity.java

```diff
@@ -0,0 +1,260 @@
+    if (version != null) {
+      titleStr += " " + version;
+    }
+    packageActivityTitle.setText(titleStr);
+    actionBar.setCustomView(packageActivityTitle);
+
+    setContentView(R.layout.activity_package_installer);
+
+    final Button installButton = (Button) findViewById(R.id.installButton);
+    final Button cancelButton = (Button) findViewById(R.id.cancelButton);
+
+    webView = (WebView) findViewById(R.id.webView);
+    webView.getSettings().setLayoutAlgorithm(WebSettings.LayoutAlgorithm.NORMAL);
+    webView.getSettings().setJavaScriptEnabled(true);
+    webView.getSettings().setUseWideViewPort(true);
+    webView.getSettings().setLoadWithOverviewMode(true);
+    webView.getSettings().setBuiltInZoomControls(true);
+    webView.getSettings().setSupportZoom(true);
+    webView.getSettings().setTextZoom(200);
+    webView.setVerticalScrollBarEnabled(true);
+    webView.setHorizontalScrollBarEnabled(true);
+    webView.setLayerType(View.LAYER_TYPE_SOFTWARE, null);
+
+    webView.setWebChromeClient(new WebChromeClient() {
+    });
+    webView.setWebViewClient(new WebViewClient() {
+      @Override
+      public void onReceivedError(WebView view, int errorCode, String description, String failingUrl) {
+      }
+
+      @Override
+      public boolean shouldOverrideUrlLoading(WebView view, String url) {
+        if (!url.toLowerCase().equals("about:blank"))
+          view.loadUrl(url);
+
+        return true;
+      }
+
+      @Override
+      public void onPageStarted(WebView view, String url, Bitmap favicon) {
+      }
+
+      @Override
+      public void onPageFinished(WebView view, String url) {
+      }
+    });
+
+    // Determine if KMP contains welcome.htm (case-insensitive) to display
+    FileFilter welcomeFilter = new FileFilter() {
+      @Override
+      public boolean accept(File pathname) {
+        if (pathname.isFile() && pathname.getName().equalsIgnoreCase("welcome.htm")) {
+          return true;
+        }
+        return false;
+      }
+    };
+
+    File[] files = tempPackagePath.listFiles(welcomeFilter);
+    if (files.length > 0 && files[0].exists() && files[0].length() > 0) {
+      webView.loadUrl("file:///" + files[0].getAbsolutePath());
+    } else {
+      String htmlString = "<div id='welcome'><H1>Package: " + pkgId + "</H1></div>";
+      webView.loadData(htmlString, "text/html; charset=utf-8", "UTF-8");
+    }
+
+    installButton.setOnClickListener(new OnClickListener() {
+      @Override
+      public void onClick(View v) {
+        try {
+          // Remove currently installed package and then install
+          File currentPackage = new File(context.getDir("data", MODE_PRIVATE).toString() +
+            File.separator + KMManager.KMDefault_AssetPackages + File.separator + pkgId);
+          if (currentPackage.exists()) {
+            FileUtils.deleteDirectory(currentPackage);
+          }
+          installedPackageKeyboards = PackageProcessor.processKMP(kmpFile);
```

Comment:
You do realize the method `PackageProcessor.processKMP` will erase the existing directory automatically for you?  I suppose the one caveat is if it's a downgrade or sidegrade... but even then, you can add the parameter `true` after `kmpFile` and have the same behavior.

(The `// Remove currently installed package` block component is redundant code.  See line 374 of PackageProcessor.java.)